### DISCUSSION TITLE
用google的yapf套件幫你的python程式排版的更符合PEP8的格式規範

### DISCUSSION THREAD

**Discussion by chia-kai-chang** [1101-GS4719-C/Python-Community-GS4719-C#844] [優秀Python套件]
在寫python的時候，為了抱持跟大家一致的python程式風格，最好遵循PEP8的格式規範：https://peps.python.org/pep-0008/

但是實際上要遵守PEP8的格式卻不太容易，還好google推出了一個yaph套件，可以幫我們在vscode的IDE環境中，快速幫.py檔自動格式優化，以滿足一致的PEP8風格。

使用方法如下：

1. 專案官網：https://github.com/google/yapf
2. 安裝yapf：`pip install yapf`
3. 在vscode裡，點開：`File > Preferences > Settings`
4. 搜尋 `python.formatting.provider`，把自動格式化工具改為 `yapf`
5. 在 .py 檔中，按下鍵盤：`Shift + Alt + F`，就可以自動格式優化你的python程式碼了~

![2022-11-14_0-20-43](https://user-images.githubusercontent.com/1768891/201532713-df63e7ac-e694-444f-990d-1d9510ee8da4.png)

## 整理前的Python Code
``` python
import numpy as np
import matplotlib.pyplot as plt
from matplotlib.widgets import RadioButtons

t = np.arange(0.0, 2.0, 0.01)
s0 = np.sin(2*np.pi*t)
s1 = np.sin(4*np.pi*t)
s2 = np.sin(8*np.pi*t)
```

## 自動整理後的Python Code
```python
import numpy as np
import matplotlib.pyplot as plt
from matplotlib.widgets import RadioButtons

t = np.arange(0.0, 2.0, 0.01)
s0 = np.sin(2 * np.pi * t)
s1 = np.sin(4 * np.pi * t)
s2 = np.sin(8 * np.pi * t)
```

**Comment by williamyen0623** [comment]
程式碼的格式優化不能說很有用，只能說必須要用，這樣才能讓協作效率提升、節省時間

**Comment by manipulateed** [comment]
我覺得其實平常打程式有些習慣的間隔或是排版都是個人的習性，也方便自己閱讀，但若要大家一同共用的話，這個套件的確能更好的解決這個問題

**Comment by ruthchen1020** [comment]
真的蠻好的，對於新手不會命名的人，還有共同協作時真的會很好用。
